biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java

```diff
@@ -283,9 +289,19 @@ public void annotation(Annotation a) throws Exception {
 				});
 			}
 		} else if (c == null) {
-			analyzer.warning(
-					"Unable to determine whether the meta annotation %s applied to type %s provides bundle annotations as it is not on the project build path. If this annotation does provide bundle annotations then it must be present on the build path in order to be processed",
-					fqn, current.getFQN());
+			// Don't repeatedly log for the same missing annotation
+			if (loggedMissing.add(fqn)) {
+				// Only issue a warning if pedantic
+				if (analyzer.isPedantic()) {
+					analyzer.warning(
+							"Unable to determine whether the meta annotation %s applied to type %s provides bundle annotations as it is not on the project build path. If this annotation does provide bundle annotations then it must be present on the build path in order to be processed",
+							fqn, current.getFQN());
+				} else {
+					analyzer.getLogger().info(
```

Comment:
Don't use the Processor's Logger since you lose the class doing the logging as the source of the logging. (We should really get rid of Processor.getLogger().) I suggest that this class create its own Logger and then use that Logger to log.